src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -442,4 +498,95 @@ private Model GetVocabularyDictionary()
             }
         }
     }
+
+    public sealed class WordEmbeddingsEstimator : TrivialEstimator<WordEmbeddingsTransform>
+    {
+        public WordEmbeddingsEstimator(IHostEnvironment env, string inputColumn, string outputColumn,
+           WordEmbeddingsTransform.PretrainedModelKind? modelKind = WordEmbeddingsTransform.PretrainedModelKind.Sswe)
+            : this(env, new WordEmbeddingsTransform(env, inputColumn, outputColumn, modelKind))
+        {
+        }
+
+        public WordEmbeddingsEstimator(IHostEnvironment env,
+            WordEmbeddingsTransform.PretrainedModelKind? modelKind = WordEmbeddingsTransform.PretrainedModelKind.Sswe,
+            string customLookupTable = null, params WordEmbeddingsTransform.Column[] columns)
+            : this(env, new WordEmbeddingsTransform(env, modelKind, customLookupTable, columns))
+        {
+        }
+
+        public WordEmbeddingsEstimator(IHostEnvironment env, WordEmbeddingsTransform transformer)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(WordEmbeddingsEstimator)), transformer)
+        {
+        }
+
+        public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            Host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.Columns.ToDictionary(x => x.Name);
+            foreach (var colInfo in Transformer.Columns)
+            {
+                if (!inputSchema.TryFindColumn(colInfo.input, out var col))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.input);
+                if (!(col.ItemType is TextType) || col.Kind != SchemaShape.Column.VectorKind.VariableVector)
```

Comment:
>VariableVector [](start = 93, length = 14)

This is a rather stronger test than the existing `TestIsTextVector` test. Variable size input vectors *can* be used of course, but also fixed sized vectors can be used. Unless there is a very strong argument for it, I'd rather not degrade our capability, since there's a fairly straightforward way to apply it to fixed size columns as well.

Reply:
That makes sense. I relaxed the checking to either be vector or variable length vector.

---
In reply to: [218167090](https://github.com/dotnet/machinelearning/pull/928#discussion_r218167090) [](ancestors = 218167090)

Reply:
Thanks @zeahmed. An extra note, while I think the *estimator* ought to reflect the full capabilities of the transform, I think it is probably fine for the pigsty extension method to remain over varvectors only.

---
In reply to: [218171027](https://github.com/dotnet/machinelearning/pull/928#discussion_r218171027) [](ancestors = 218171027,218167090)